core/trino-main/src/main/java/io/trino/Session.java

```diff
@@ -867,7 +867,7 @@ public Session build()
                     Optional.ofNullable(clientInfo),
                     clientTags,
                     clientCapabilities,
-                    Optional.ofNullable(resourceEstimates).orElse(new ResourceEstimateBuilder().build()),
+                    Optional.ofNullable(resourceEstimates).orElseGet(() -> new ResourceEstimateBuilder().build()),
```

Comment:
The `ResourceEstimateBuilder` has 3 optionals so I guess this one may stay

Reply:
i reverted this one based on the assumption that `resourceEstimates` is almost never set, so we will go the Else branch anyway